march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -251,198 +255,236 @@ void MarchHardwareInterface::read(ros::Duration elapsed_time)
 
 void MarchHardwareInterface::write(ros::Duration elapsed_time)
 {
-  positionJointSoftLimitsInterface.enforceLimits(elapsed_time);
-
-  for (int i = 0; i < num_joints_; i++)
-  {
-    if (marchRobot.getJoint(joint_names_[i]).canActuate())
+    for (int i = 0; i < num_joints_; i++)
     {
-      ROS_DEBUG("After limits: Trying to actuate joint %s, to %lf rad, %f "
-                "speed, %f effort.",
-                joint_names_[i].c_str(), joint_position_command_[i], joint_velocity_command_[i],
-                joint_effort_command_[i]);
-      marchRobot.getJoint(joint_names_[i]).actuateRad(static_cast<float>(joint_position_command_[i]));
+        ROS_DEBUG("Before limits: Trying to actuate joint %s, to %lf "
+                  "rad, %f speed, %f effort.",
+                  joint_names_[i].c_str(), joint_position_command_[i], joint_velocity_command_[i],
+                  joint_effort_command_[i]);
     }
-  }
 
-  if (hasPowerDistributionBoard)
-  {
-    updatePowerDistributionBoard();
-  }
-}
+    positionJointSoftLimitsInterface.enforceLimits(elapsed_time);
 
-void MarchHardwareInterface::resetIMotionCubesUntilTheyWork()
-{
-  bool encoderSetCorrectly = false;
-
-  while (!encoderSetCorrectly)
-  {
-    encoderSetCorrectly = true;
-    for (int i = 0; i < num_joints_; ++i)
+    for (int i = 0; i < num_joints_; i++)
     {
-      march4cpp::Joint joint = marchRobot.getJoint(joint_names_[i]);
-      if (joint.getAngleIU() == 0)
-      {
-        ROS_ERROR("Joint %s failed (encoder reset)", joint_names_[i].c_str());
-        encoderSetCorrectly = false;
-      }
+        march4cpp::Joint singleJoint = marchRobot.getJoint(joint_names_[i]);
+
+        if (marchRobot.getJoint(joint_names_[i]).canActuate())
+        {
+            ROS_DEBUG("After limits: Trying to actuate joint %s, to %lf rad, %f "
+                      "speed, %f effort.",
+                      joint_names_[i].c_str(), joint_position_command_[i], joint_velocity_command_[i],
+                      joint_effort_command_[i]);
+            marchRobot.getJoint(joint_names_[i]).actuateRad(static_cast<float>(joint_position_command_[i]));
+        }
```

Comment:
Please refactor this to use the style develop uses at the moment.
```
march4cpp::Joint joint = marchRobot.getJoint(joint_names_[i]);
```
Once and use the variable later